RecoTauTag/RecoTau/plugins/PATTauDiscriminationByMVAIsolationRun2.cc

```diff
@@ -209,6 +212,17 @@ double PATTauDiscriminationByMVAIsolationRun2::discriminate(const TauRef& tau) c
     // ---
     float leadingTrackChi2 = tau->leadingTrackNormChi2();
     float eRatio = clusterVariables_.tau_Eratio(*tau);
+
+    // Difference between measured and maximally allowed Gottfried-Jackson angle
+    float gjAngleDiff = -999;
+    if ( tauDecayMode == 10 ) {
+        double mTau = 1.77682;
+        double mAOne = tau->p4().M();
+        double pAOneMag = tau->p();
+        double thetaGJmax = TMath::ASin( (TMath::Power(mTau,2) - TMath::Power(mAOne,2) ) / ( 2 * mTau * pAOneMag ) );
+        double thetaGJmeasured = TMath::ACos( ( tau->p4().px() * decayDistX + tau->p4().py() * decayDistY + tau->p4().pz() * decayDistZ ) / ( pAOneMag * decayDistMag ) );
```

Comment:
Dear Slava,

the valua of `mAOne` can only be between 0.8 and 1.5 GeV. This cut is placed on 3-prong taus during the tau reconstruction by [HPS](https://github.com/cms-sw/cmssw/blob/master/RecoTauTag/RecoTau/python/PFRecoTauDiscriminationByHPSSelection_cfi.py#L86-L87). Thus, there is no need for protection.

Cheers,
Alex